dataprofiler/tests/profilers/profiler_options/test_unstructured_text_profile_options.py

```diff
@@ -0,0 +1,178 @@
+            if prop not in ['words', 'vocab']:
+                option.set({prop: value})
+                self.assertEqual(value, getattr(option, prop), msg=prop)
+            else:
+                prop_enable = '{}.is_enabled'.format(prop)
+                option.set({prop_enable: value})
+                self.assertEqual(value, option.properties[prop].is_enabled,
+                                 msg=prop)
+
+        for params in params_to_check:
+            prop, value_list = params['prop'], params['value_list']
+            for value in value_list:
+                _assert_set_helper(prop, value)
+
+        # Treat is_case_sensitive and stop_words as BooleanOption
+        expected_error = ("type object 'is_case_sensitive' has no attribute "
+                          "'is_enabled'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option.set({'is_case_sensitive.is_enabled': True})
+
+        expected_error = ("type object 'stop_words' has no attribute "
+                          "'is_enabled'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option.set({'stop_words.is_enabled': True})
+
+        expected_error = ("type object 'words.is_enabled' has no attribute "
+                          "'other_props'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option.set({'words.is_enabled.other_props': True})
+
+        expected_error = ("type object 'vocab.is_enabled' has no attribute "
+                          "'other_props'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option.set({'vocab.is_enabled.other_props': True})
+
+    def test_validate_helper(self):
+        super(TestTextProfilerOptions, self).test_validate_helper()
+
+    def test_validate(self):
+
+        super(TestTextProfilerOptions, self).test_validate()
+
+        params_to_check = [
+            # non errors
+            dict(prop='is_enabled', value_list=[False, True], errors=[]),
+            dict(prop='is_case_sensitive', value_list=[False, True], errors=[]),
+            dict(prop='stop_words',
+                 value_list=[None, ['word1', 'word2'], []],
+                 errors=[]),
+            dict(prop='words',
+                 value_list=[BooleanOption(is_enabled=False),
+                             BooleanOption(is_enabled=True)],
+                 errors=[]),
+            dict(prop='vocab',
+                 value_list=[BooleanOption(is_enabled=False),
+                             BooleanOption(is_enabled=True)],
+                 errors=[]),
+
+            # errors
+            dict(prop='is_case_sensitive',
+                 value_list=[2, 'string'],
+                 errors=[
+                     "TextProfilerOptions.is_case_sensitive must "
+                     "be a Boolean."
+                 ]),
+            dict(prop='stop_words',
+                 value_list=[2, 'a', [1, 2]],
```

Comment:
we should also add a test for mixed list e.g. [`1, 'a']`

Reply:
added